src/Microsoft.AspNetCore.Session/DistributedSession.cs

```diff
@@ -194,58 +198,68 @@ private void Load()
         }
 
         // This will throw if called directly and a failure occurs. The user is expected to handle the failures.
-        public async Task LoadAsync()
+        public async Task LoadAsync(CancellationToken cancellationToken)
         {
+            cancellationToken.ThrowIfCancellationRequested();
             if (!_loaded)
             {
-                var data = await _cache.GetAsync(_sessionKey);
-                if (data != null)
+                using (var timeout = new CancellationTokenSource(_ioTimeout))
                 {
-                    Deserialize(new MemoryStream(data));
-                }
-                else if (!_isNewSessionKey)
-                {
-                    _logger.AccessingExpiredSession(_sessionKey);
+                    var cts = CancellationTokenSource.CreateLinkedTokenSource(timeout.Token, cancellationToken);
+                    var data = await _cache.GetAsync(_sessionKey, cts.Token);
+                    if (data != null)
```

Comment:
True, but it would mess up the scope of the data variable. Deserialization is quick enough that I'm not worried about it.